### PR TITLE
add application inbound port as a service

### PR DESCRIPTION
**Pull request by FernetMenta**
Add App inbound port as a service. Allows components like windowing or lirc to send events to CApp without using global g_application.
Use shared pointer for registration of serivce allows safe destruction.

### CODE REVIEW

**Comment by FernetMenta** [comment]
@garbear this impolements what we discussed on Lirc PR

**Comment by garbear** [comment]
Awesometastic. This is exactly what we needed. thanks!

**Comment by FernetMenta** [comment]
> why only OSX is affected?

other platforms need to be done.

File: xbmc/windowing/android/WinEventsAndroid.cpp [hunk 1]
```diff
@@ -106,7 +106,9 @@ bool CWinEventsAndroid::MessagePump()
       m_events.pop_front();
     }
 
-    ret |= g_application.OnEvent(pumpEvent);
+    std::shared_ptr<CAppInboundProtocol> appPort = CServiceBroker::GetAppPort();
+    if (appPort)
+      ret |= appPort->OnEvent(newEvent);
```
Comment (Rechi) [inline xbmc/windowing/android/WinEventsAndroid.cpp]: `appPort->OnEvent(pumpEvent);`

**Review by afedchin — APPROVED** [review]
windows part looks good.

**Review by peak3d — APPROVED** [review]
Runtime tested with AML aarch64 gmb build

**Pull request merged**